### ISSUE TITLE
[Bug]: Desktop App: Server is not running at this URL. 

### ISSUE THREAD

**Issue by atrueresistance** [actualbudget/actual#3304]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

I'm having an issue with the web server connecting to the desktop application. 

![image](https://github.com/user-attachments/assets/670eba51-55cb-4b9a-99c4-cf3f1cf9eac1)

The server is up and running using HTTPS and a self signed cert. 
![image](https://github.com/user-attachments/assets/1a229ecf-0ebf-40e7-b7aa-8b249da1823f)

Having a self signed cert, you would normally be prompted in Chrome stating that the certificate is not valid. It's probably not the best way to go about it, but if you add it to the Trusted Root Certification Authorities for the current user Chrome inherently trusts it solving the warning. At least on the local machine for the current logged on user. I don't intend on opening the server to the outside, just LAN connection is good enough for me.
![image](https://github.com/user-attachments/assets/0f7cbd1f-8268-4bb6-b731-8edd7fa322bb)

The desktop application is being ran on the same machine and user that the certificate was added to the Trusted Root Certification Authorities store.

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Desktop App (Electron)

### Operating System

Windows 11

**Comment by MikesGlitch** [comment]
Confirmed - I'm seeing this also. This may be tricky - I'm not sure we want to turn off TLS validation for security reasons.

However, disabling it does work:
```
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
```

Anyone got any thoughts on this?

**Comment by psybers** [comment]
https://stackoverflow.com/questions/38986692/how-do-i-trust-a-self-signed-certificate-from-an-electron-app

**Comment by atrueresistance** [comment]
I'll start of with I know nothing of electron apps. I do agree that TLS validation off is not the default way to ship. 

I'm not sure how much work it would be, but would it be possible to have a configuration item in the desktop app that whitelists a specific certificate serial number? 

https://www.electronjs.org/docs/latest/api/session#sessetcertificateverifyprocproc 

If the certificate is installed in the current user's certificate store can the desktop app use is directly from the store?

**Comment by MikesGlitch** [comment]
We have a bit of an odd setup, we run our loot-core server process (that calls the API and gets the error) in a forked process. 

I think that means the session's setCertificateVerifyProc won't pick it up. 

Maybe we can configure node-fetch for the electron to figure out the root store 🤔

**Comment by MikesGlitch** [comment]
I have a draft pr up - as expected the electron setCertificateVerifyProc wont work. The only way to make it work is for the user to specify the cert that they want to trust. 

@atrueresistance How did you generate your cert? I used makecert to test but want to make sure it also works on your config.

**Comment by atrueresistance** [comment]
@MikesGlitch I used the same command as in the documentation except with a subject alternative name to match the hostname. 

`openssl req -x509 -nodes -days 365 -addext "subjectAltName = DNS:orangepi3-lts" -newkey rsa:2048 -keyout selfhost.key -out selfhost.crt`

**Comment by MikesGlitch** [comment]
> @MikesGlitch I used the same command as in the documentation except with a subject alternative name to match the hostname.

Thanks, I have a build here: https://github.com/actualbudget/actual/actions/runs/10541997136/job/29208287390?pr=3308

Can you confirm if it fixes this issue?

**Comment by atrueresistance** [comment]
`This app package’s publisher certificate could not be verified. Contact your system administrator or the app developer to obtain a new app package with verified certificates. The root certificate and all immediate certificates of the signature in the app package must be verified (0x800B010A)`

**Comment by MikesGlitch** [comment]
That's what we'd expect for the appx installer - it's for the windows store. 

Try the "actual-electron-windows-latest" artifact, that should install as per usual.

**Comment by atrueresistance** [comment]
Is there a link? Sorry I'm not sure where I should be looking.

**Comment by MikesGlitch** [comment]
> Is there a link? Sorry I'm not sure where I should be looking.

Ah, this is the direct link to the download: https://github.com/actualbudget/actual/actions/runs/10541997136/artifacts/1850858005

Here's the PR, if interested: https://github.com/actualbudget/actual/pull/3308

**Comment by crzdg** [comment]
Nice. I will soon test this as well. Im using Caddy as proxy, so a short-lived certificate. Can I trust the Intermediate / Root, so the short-lived signed certs work?

**Comment by MikesGlitch** [comment]
> Nice. I will soon test this as well. Im using Caddy as proxy, so a short-lived certificate. Can I trust the Intermediate / Root, so the short-lived signed certs work?

I'm not too familiar with Caddy so couldn't say. You can trust a certificate that is locally on the PC that is .crt or .pem format.

**Comment by MikesGlitch** [comment]
Closed by https://github.com/actualbudget/actual/pull/3308

**Comment by wgregbrown** [comment]
Is there anywhere to still get the desktop client that will work with self signed certs. @MikesGlitch put a link

https://github.com/actualbudget/actual/actions/runs/10541997136/job/29208287390?pr=3308

But its empty now.

**Comment by MikesGlitch** [comment]
The latest release has it. You can get it from the GitHub releases page

**Comment by neobladerunner** [comment]
I'm experiencing this issue with latest version 26.3.0 via Flathub. I've generated a self-signed cert via mkcert and can load the app fine via the browser from any device on my network. The desktop app states:

> Server is not running at this URL. Make sure you have HTTPS set up properly.
> If the server is using a self-signed certificate select it here.

I followed the instructions [here](https://actualbudget.org/docs/config/https/) to import the root certificate to the desktop app from the device Actual is running on, but the UI doesn't seem to do anything when I import it. Clicking on OK doesn't clear the red error message, and I can't tell if it's rejecting the cert file or not seeing it at all. I suppose I should just use the web client but wanted to try the desktop app if I could.
